data-prepper-plugins/aggregate-processor/src/test/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateGroupManagerTest.java

```diff
@@ -8,23 +8,29 @@
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
+import java.time.Instant;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
+import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.sameInstance;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.Matchers.is;
-import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class AggregateGroupManagerTest {
 
     private AggregateGroupManager aggregateGroupManager;
 
     private AggregateIdentificationKeysHasher.IdentificationHash identificationHash;
 
+    private static final int TEST_WINDOW_DURATION = 10;
```

Comment:
I'd recommend testing with a random window duration which falls into some known window. Maybe generated from 10-20? It would help prevent issues where the source code supports only the current test input. But, it's not necessary.